src/Idris/Parser.idr

```diff
@@ -776,23 +776,50 @@ mutual
   expr : ParseOpts -> FileName -> IndentInfo -> Rule PTerm
   expr = typeExpr
 
+  interpBlock : ParseOpts -> FileName -> IndentInfo -> Rule PTerm
+  interpBlock q fname idents = interpBegin *> (mustWork $ expr q fname idents) <* interpEnd
+
   export
-  interpStr : ParseOpts -> FileName -> IndentInfo -> Rule (List PStr)
-  interpStr q fname idents = do
-      strBegin
-      commit
-      xs <- many $ bounds $ interpBlock <||> strLit
-      strEnd
-      pure $ toPString <$> xs
+  singlelineStr : ParseOpts -> FileName -> IndentInfo -> Rule PTerm
+  singlelineStr q fname idents
+      = do b <- bounds $ do strBegin
+                            commit
+                            xs <- many $ bounds $ (interpBlock q fname idents) <||> strLitLines
+                            strEnd
+                            the (SourceEmptyRule _) $
+                                case traverse toPStr xs of
+                                     Left err => fatalError err
+                                     Right pstrs => pure $ pstrs
+           pure $ PString (boundToFC fname b) b.val
     where
-      interpBlock : Rule PTerm
-      interpBlock = interpBegin *> (mustWork $ expr q fname idents) <* interpEnd
+      toPStr : (WithBounds $ Either PTerm (List1 String)) -> Either String PStr
+      toPStr x = case x.val of
+                      Right (str:::[]) => Right $ StrLiteral (boundToFC fname x) str
+                      Right (_:::strs) => Left "Multi-line string is expected to begin with \"\"\""
```

Comment:
But since this error is caught in the parser, it'll be hard to move the error point backward. Besides, this error points to the first wrapped line of a single-line string, so I think it may be reasonable?

Reply:
Another solution is to postpone the check to `desugar`.

Reply:
Either way is fine, if we don't want to deal with it now we can just open an issue just like for the "FIXME"